### DISCUSSION TITLE
A new AsyncPlayerSendCommandsEvent with CommandSourceStack instead of BukkitBrigadierCommandSource

### DISCUSSION THREAD

**Discussion by davidmayr** [PaperMC/Paper#10783] [Ideas]
### Is your feature request related to a problem?

BukkitBrigadierCommandSource is now deprecated and scheduled for removal. Although AsyncPlayerSendCommandsEvent isn't scheduled for removal or deprecated, I would assume it would fall with it.

I rely on this event since I have a system where players on the same server are separated into different games with different commands. Some commands may exist in both games, some don't. Some may have the same name as others but fully different arguments and functionality. Player#updateCommands() is used when switching games. They are then handled in the PlayerCommandPreprocessEvent. 
Therefor, I cannot register these commands globally and instead append them in this event. 

### Describe the solution you'd like.

A new event (or a breaking update to the current one) adapting the new interface (that is not deprecated or scheduled for removal) with the same functionality. 

### Describe alternatives you've considered.

Using protocol lib to manipulate the packets

**Comment by jpenilla** [comment]
#10826 

Due to specific circumstances, this is only a breaking API change, old plugins are still ABI compatible without needing any bytecode mangling.
